### ISSUE TITLE
why comment the frcnn_proposal_layer's Forward GPU implementation？

### ISSUE THREAD

**Issue by sigfrid696** [makefile/frcnn#18]
Hello,

I understand that this is a quite old repo, but I would like to know if the GPU implementation has some bug, since it is commented out.

Thanks

**Comment by sigfrid696** [comment]
I did some tests after porting the implementation to my library...the implementation seems ok because it produces the same results as obtained by the proposal layer done fully on CPU. The only problem is that is significantly slower

**Comment by sigfrid696** [comment]
Sorry I accidentally closed the issue...After doing some tests I found that only gpu nms processing of the proposal layer makes the overall process faster, the other parts of the layer remaining implemented on CPU...hope these conclusions can be of any help, if anyone is interested in the same algorithm

**Comment by makefile** [comment]
Yes, the GPU implementation in frcnn_proposal_layer.cu is not that efficient.

**Comment by sigfrid696** [comment]
Ok , thanks for the reply. After studying a bit the code, it doesn't seem related to the fact that is not an optimized algorithm. Probably some parts of the layer can be executed better on CPU, other on GPU, as the nms part, which is somehow similar to a convolution for the operations involved

**Comment by makefile** [comment]
Yes, you are right. The Soft-NMS operation is similar with this.
